piksi_tools/stm_unique_id.py

```diff
@@ -14,36 +14,51 @@
 import sys
 import serial_link
 
+from piksi_tools.heartbeat import Heartbeat
+
 from sbp.flash          import SBP_MSG_STM_UNIQUE_ID_DEVICE
 from sbp.client.handler import *
 
-class STMUniqueID:
+class STMUniqueID(object):
+  """
+  Retrieve the STM Unique ID from Piksi.
+  """
 
   def __init__(self, link):
-    self.heartbeat_received = False
+    """
+    Parameters
+    ==========
+    link : sbp.client.handler.Handler
+      link to register Heartbeat message callback with
+    """
     self.unique_id_returned = False
-    # SBP version is unset in older devices.
-    self.sbp_version = (0, 0)
     self.unique_id = None
     self.link = link
-    link.add_callback(self.receive_heartbeat, SBP_MSG_HEARTBEAT)
+    self.heartbeat = Heartbeat(link)
     link.add_callback(self.receive_stm_unique_id_callback, SBP_MSG_STM_UNIQUE_ID_DEVICE)
```

Comment:
Yeah, this is a case where you wouldn't use the context handling as the Heartbeat object has to persist in between STMUniqueID calls, and so you need to clean up upon exit (this should get added here). What I'm more concerned about are scripts where you may be constantly creating/destroying the Heartbeats. I'll change it though.